packages/spor-alert-react-native/src/Alert.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  composeRestyleFunctions,
+  createVariant,
+  spacing,
+  SpacingProps,
+  spacingShorthand,
+  SpacingShorthandProps,
+  useRestyle,
+  useTheme,
+  VariantProps,
+} from "@shopify/restyle";
+import {
+  CloseOutline18Icon,
+  DropdownDownFill18Icon,
+  DropdownUpFill18Icon,
+} from "@vygruppen/spor-icon-react-native";
+import React, { useEffect, useState } from "react";
+import { Box } from "@vygruppen/spor-layout-react-native";
+import { Theme } from "@vygruppen/spor-theme-react-native";
+import { Text } from "@vygruppen/spor-typography-react-native";
+import { Linking, Pressable, TouchableOpacity } from "react-native";
+import { UrlItem } from "./UrlItem";
+
+type Variant = VariantProps<Theme, "alertColorScheme", "colorScheme">;
+const colorScheme = createVariant({
+  themeKey: "alertColorScheme",
+  property: "colorScheme",
+});
+
+type RestyleProps = SpacingProps<Theme> &
+  SpacingShorthandProps<Theme> &
+  Variant;
+
+const restyleFunctions = composeRestyleFunctions<Theme, RestyleProps>([
+  spacing,
+  spacingShorthand,
+  colorScheme,
+]);
+
+type ColorVariants =
+  | "yellow"
+  | "light-yellow"
+  | "orange"
+  | "red"
+  | "green"
+  | "blue";
+
+type ActionType = "expandable" | "closeable" | "none";
+
+type BaseProps = Exclude<RestyleProps, "variant"> & {
+  children: string;
+  colorScheme: ColorVariants;
+  url?: string;
+  icon: JSX.Element;
+  onToggle?: (isExpanded: boolean) => void;
+};
+
+type WithExpandableProps = {
+  actionType: ActionType;
+  title: string;
+};
+
+type WithCloseButtonProps = {
+  actionType: ActionType;
+  title: string;
+  onPress: () => void;
```

Comment:
Jeg synes denne burde hete onClose - så man er litt ekstra eksplisitt.

Man _kan_ jo også vurdere å droppe `actionType` til fordel for å se om man har spesifisert `onClose` og evnt `onToggle` / `children`. Hva synes dere andre?

Dvs at man har følgende versjoner:

```tsx
<Alert onClose={handleClose} title="tittel" /> // med lukkeknapp
<Alert onToggle={handleToggle} title="tittel"> // expandable
  <Text>Ekspanderbart innhold</Text>
</Alert>
<Alert title="tittel" /> // basic
```

Alternativt kan man lage egne komponenter for de tre, for å gjøre implementasjonen litt mer oversiktlig og sammensatt:

```tsx
<SimpleAlert title="Some title" />
<ClosableAlert onClose={handleClose} title="Some title" />
<ExpandableAlert 
  onToggle={handleToggle} 
  isExpanded={isExpanded} 
  defaultExpanded={false} 
  title="Some title"
>
  <Text>Expandable content</Text>
</ExpandableAlert>
```

Man kunne delt utseende i en intern `BaseAlert` komponent, som brukere ikke får direkte tilgang til f.eks.?

Mange forskjellige approaches om dere vil utforske dem.

Reply:
Prøvde meg litt fram, kom fram til at med en BaseAlert blir brukergrensesnittet enklere? Pusha et utkast😊